src/file/path.rs

```diff
@@ -0,0 +1,160 @@
+    pub lines: Vec<Line>,
+    /// Original path of ssh_config file.
+    ///
+    /// It is used to be able to restore the source of entries provided by this file.
+    pub path: Option<PathBuf>,
+}
+
+impl fmt::Display for File {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            self.lines
+                .iter()
+                .map(ToString::to_string)
+                .collect::<Vec<String>>()
+                .join("")
+        )
+    }
+}
+
+/// Line is a sequence of characters in ssh_config file followed by /n or /r/n.
+#[derive(PartialEq, Debug)]
+pub struct Line {
+    /// Indent prefix is the longest possible line prefix consisting of whitespace symbols.
+    ///
+    /// It is convenient to keep it to be able to restore original line formatting.
+    pub indent_prefix: WhitespaceString,
+    /// Expression is an optional sequence of characters between [`indent_prefix`] and [`indent_suffix`].
+    ///
+    /// See [`Expression`] for details.
+    pub expression: Expression,
+    /// Indent suffix is the longest possible line suffix consisting of whitespace symbols, not intersecting with [`indent_prefix`].
+    ///
+    /// Though keeping [`indent_prefix`] may seem useless, it is stored for the purpose of idempotent
+    /// reading and writing ssh_config files.
+    pub indent_suffix: WhitespaceString,
+}
+
+impl fmt::Display for Line {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}{}{}\n",
+            self.indent_prefix, self.expression, self.indent_suffix
+        )
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub enum Expression {
+    /// Configuration options is an entry of a keyword-argument format.
+    ConfigurationOptions {
+        /// Keyword is a case-insensitive token consisting of `[A-Za-z]` and corresponding to the field being set.
+        keyword: String,
+        /// Separator is a string splitting keyword and arguments.
+        ///
+        /// Separator can be represented exactly in two different ways:
+        /// - an arbitrary sequence of whitespace symbols (e.g., `Keyword argument`);
+        /// - equals (`=`) sign preceded or/and followed by an arbitrary sequences of whitespace symbols (e.g. `Keyword=argument`, `Keyword = argument`).
+        separator: String,
+        /// Arguments expression is a sequence of argument tokens.
+        ///
+        /// See [`ArgumentTokenSequence`] for details.
+        arguments_expression: ArgumentTokenSequence,
+    },
+    /// Comment is a string starting with hash (`#`) symbol.
+    ///
+    /// Please, note that genuine understanding of comment lines in [ssh_config(5)](https://linux.die.net/man/5/ssh_config)
+    /// is different from what we call a comment, because we distinguish [`Empty`] expression as a separate case, not as a comment.
+    Comment(String),
```

Comment:
Do we really need to split these definitions? Maybe we can do it like man says and then split these two cases later, not now. Here we seem to want just parsing ssh_config file, without inventing a bicycle